### DISCUSSION TITLE
Faker REST API

### DISCUSSION THREAD

**Discussion by RayyanNafees** [faker-js/faker#2272] [Ideas]
## Inspiration: JSON-Placeholder

How about a faker REST api much like that of [json-placeholder](https://jsonplaceholder.typicode.com) to get fake data
```js
fetch('https://jsonplaceholder.typicode.com/posts/1')
```

## Faker Example
In a similar way faker can also implement a REST API by using the  library's **methods** as REST **endpoints**

For example, the following fakerjs code
```js
faker.git.branch() // 'feed-parse'
```

will return the same data as fetching the endpoint:
```js
fetch('https://fakerjs.vercel.app/git/branch')
```

**Comment by ST-DDT** [comment]
Thanks for your suggestions, what would be your intended usecase?
IMO a generic web service cannot produce the data that I would need for my services.

Do you know of a web server provider that would affordably host this service for an OSS project?

Or are you referring to having a template project to host such an API?

**Comment by matthewmayer** [comment]
I'm not sure this really needs to be part of the official @faker-js distribution. It's pretty trivial to make a basic web server using your preferred programming language. For example here's an toy implementation in 12 lines using Express:
```js
const express = require('express');
const {faker} = require('@faker-js/faker');
const app = express();
const port = 8888;
app.get('/:module/:method', (req, res) => {
    try {
        res.json(faker[req.params.module][req.params.method]());
    } catch (err) {
        res.status(500).json({error: "Not a valid method call"});
    }
})
app.listen(port, () => console.log(`Listening on port ${port}`));
```
But to go from this to a full implementation would require a lot of decisions about the structure of the API which are likely to be implementation-specific, as well as figuring out lots of things about hosting.
* how would you request multiple values?
* how would you set a seed/handle reproducible results?
* how would you pass parameters?
* could you pass a template like with faker.helpers.fake?
* how do you handle errors?
* do you need rate-limiting?
* do you need api keys?
* how do you prevent abuse of the service?
* who pays for hosting?
* what domain name to use?
* do you need server monitoring?
* how do you handle versioning of the underlying Faker library?

Realistically most use-cases are much better served by either using the NPM package locally (much faster) or setting up your own server which returns data in the format you need.

**Comment by sanjaykhanssk** [comment]
you can checkout [fakend.fyi](https://fakend.fyi)
